src/main/java/frc/trigon/robot/subsystems/roller/Roller.java

```diff
@@ -0,0 +1,46 @@
+package frc.trigon.robot.subsystems.roller;
+
+
+import frc.trigon.robot.subsystems.MotorSubsystem;
+import org.littletonrobotics.junction.Logger;
+
+public class Roller extends MotorSubsystem {
+    private final static Roller INSTANCE = new Roller();
+    private final RollerIO rollerIO = RollerIO.generateIO();
+    private final RollerInputsAutoLogged rollerInputs = new RollerInputsAutoLogged();
+
+    public static Roller getInstance() {
+        return INSTANCE;
+    }
+
+    private Roller() {
+    }
+
+    @Override
+    public void stop() {
+        rollerIO.stopMotor();
+    }
+
+    @Override
+    public void periodic() {
+        rollerIO.updateInputs(rollerInputs);
+        Logger.processInputs("Roller", rollerInputs);
+        if (isInfraredSensorTriggered()) {
+            stop();
+            setTargetState(RollerConstants.RollerState.DEFAULT);
+        }
```

Comment:
You want this to happen only if the infrared sensor is triggered and the target state is collected.